### ISSUE TITLE
Eigen 3.3 support

### ISSUE THREAD

**Issue by ahoarau** [ocra-recipes/eigen_lgsm#2]
Currently lgsm does not support Eigen <3.3 (beta2 present in 16.04 by default) as fails to compile.

**Comment by traversaro** [comment]
Out of curiosity, I tried to run `git bisect` and apparently the first commit for which the test is not compiling is https://github.com/RLovelett/eigen/commit/8668c9f3828ad976e760c9081164ca01734b2124 .

**Comment by traversaro** [comment]
I am state the obvious, but this is related to the new expression evaluators implemented in Eigen 3.3 :  http://eigen.tuxfamily.org/index.php?title=3.3#Expression_evaluators .

**Comment by ahoarau** [comment]
Yep I tried to see what was needed https://github.com/ocra-recipes/eigen_lgsm/commit/9e0cedc923a114d38e2f62400895f48b6fcf72b6
 But I guess it's too much work/understanding for such a tiny lib..
